lapis/src/test/kotlin/org/genspectrum/lapis/model/AdvancedQueryFacadeTest.kt

```diff
@@ -0,0 +1,774 @@
+
+        assertThat(result, equalTo(NucleotideInsertionContains(1234, "G.*A.*G", null)))
+    }
+
+    @Test
+    fun `given amino acid mutation expression then should map to AminoAcidSymbolEquals`() {
+        val advancedQuery = "S:N501Y"
+
+        val result = underTest.map(advancedQuery)
+
+        assertThat(result, equalTo(AminoAcidSymbolEquals("S", 501, "Y")))
+    }
+
+    @Test
+    fun `given amino acid mutation expression with lower case letters then should map to AminoAcidSymbolEquals`() {
+        val advancedQuery = "S:n501y"
+
+        val result = underTest.map(advancedQuery)
+
+        assertThat(result, equalTo(AminoAcidSymbolEquals("S", 501, "Y")))
+    }
+
+    @Test
+    fun `given amino acid mutation expression with gene lower case letters then should map to AminoAcidSymbolEquals`() {
+        val advancedQuery = "orf1a:N501Y"
+
+        val result = underTest.map(advancedQuery)
+
+        assertThat(result, equalTo(AminoAcidSymbolEquals("ORF1a", 501, "Y")))
+    }
+
+    @Test
+    fun `given amino acid mutation expression without first symbol then should map to AminoAcidSymbolEquals`() {
+        val advancedQuery = "S:501Y"
+
+        val result = underTest.map(advancedQuery)
+
+        assertThat(result, equalTo(AminoAcidSymbolEquals("S", 501, "Y")))
+    }
+
+    @Test
+    fun `given amino acid mutation expression without second symbol then should return HasAminoAcidMutation`() {
+        val advancedQuery = "S:N501"
+
+        val result = underTest.map(advancedQuery)
+
+        assertThat(result, equalTo(HasAminoAcidMutation("S", 501)))
+    }
+
+    @Test
+    fun `given amino acid mutation expression without any symbol then should return HasAminoAcidMutation`() {
+        val advancedQuery = "S:501"
+
+        val result = underTest.map(advancedQuery)
+
+        assertThat(result, equalTo(HasAminoAcidMutation("S", 501)))
+    }
+
+    @Test
+    fun `given a valid advancedQuery with a 'AA insertion' expression then returns SILO query`() {
+        val advancedQuery = "ins_S:501:EPE"
+
+        val result = underTest.map(advancedQuery)
+
+        assertThat(result, equalTo(AminoAcidInsertionContains(501, "EPE", "S")))
+    }
+
+    @Test
+    fun `given a valid advancedQuery with a stop codon expression then returns SILO query`() {
+        val advancedQuery = "ins_S:501:A*C"
+
+        val result = underTest.map(advancedQuery)
+
+        assertThat(result, equalTo(AminoAcidInsertionContains(501, "A\\*C", "S")))
+    }
+
+    @Test
+    fun `given a valid advancedQuery with a 'AA insertion' expression with lower case then returns SILO query`() {
+        val advancedQuery = "ins_ORF1a:501:ePe"
+
+        val result = underTest.map(advancedQuery)
+
+        assertThat(result, equalTo(AminoAcidInsertionContains(501, "EPE", "ORF1a")))
+    }
+
+    @Test
+    fun `given an invalid advancedQuery with an invalid gene return error`() {
+        val advancedQuery = "ins_invalidGene:501:EPE"
+
```

Comment:
This test is duplicate. We already have `given a advancedQuery with a 'Insertion' expression with invalid sequenceName throws`.

Instead we're missing a test that checks an invalid gene for a mutation query.